submariner/centralized-control-plane.md

```diff
@@ -0,0 +1,236 @@
+# Centralized Service Control Plane
+
+## Summary
+
+Current Submariner service management follows the model and guidelines
+ defined in [MCS](https://github.com/kubernetes-sigs/mcs-api). Specifically the
+ current design has these attributes:
+
+- Exported services are automatically reflected into all clusters in the ClusterSet;
+- Service are exported and imported using the same namespace and name; and
+- Service exports are managed from workload clusters.
+
+The above works well in an environment where clusters are used by a single
+ administrative domain and services are commonly shared across all clusters.
+ For example, when a company runs clusters as the runtime infrastructure and developers can deploy
+ to any cluster for availability, redundancy, or geographic proximity. Another
+ common set-up is clusters under different administrative domains (e.g., separated
+ by team). In those environments, service naming and sharing may be controlled
+ differently.
+ We would like to propose a different approach for Service management, that allows:
+
+1. Independent service naming (i.e., allow use of different names in different clusters).
+1. Selective imports (i.e., import services into a subset of clusters in the ClusterSet).
+1. Centralized control over exporting and importing (i.e., defined in Broker, not
```

Comment:
I'll try and clarify.

I view the MCS API is a building block (i.e., a mechanism) and as such can be used by a different policy engines to drive imports and exports from a central location. This might be "import everywhere" as in Submariner, or a more selective TBD policy.

What's missing, IMHO, are:
1. name independence across clusters;
1. the propagation of Exports and Import from/to workload clusters is implementation specific. Some synchronization/distribution mechanism is required (e.g., Submariner's Lighthouse is an example). I don't think this is in scope for MCS API; and
1. Policy objects (e.g., how to declaratively define which exports to import where and with what local workload clusters attributes) are similarly implementation specific and possibly out of MCS scope (for now?).

The first (name independence) seems to me a fundamental assumption on the MCS API. This makes `ServiceExport` and `ServiceImport` less useful as the abstraction in the central shared location. For example, since all imports are created in the Submariner broker namespace of the ClusterSet, Submariner resorts to tagging the destination namespace and name on the `ServiceImport` using labels and annotations. It might have been clearer to explicitly set those in `ServiceImport.Spec`. Similarly, the assumption on name sameness has implication on `ServiceExport` (e.g., two completely independent cluster, owned and managed by different teams, must agree on using unique namespaces and/or names, lest their exports conflate when shared).

While I think that the latter two (synchronization and policies) are possible enhancements to the MCS API, I also think it would be worthwhile to experiment with different approaches before attempting to reach consensus there.

Reply:
Thanks for these details, they do help clarify things! And yes it'll be great to add them to the doc eventually, thanks!

> While I think all three (naming, synchronization and policies) are possible enhancements to the MCS API, I also think it would be worthwhile to experiment with different approaches before attempting to reach consensus there.

Yes I think we'll have to show successful experimentation before reaching consensus with the SIG, but I think it'll be less work and a better result if we build that consensus in conjunction with the implementation. I shadow many of the SIG meetings, but @skitt has worked with the SIG a lot and might be interested in helping start that discussion.